webpack.config.js

```diff
@@ -0,0 +1,53 @@
+"use strict";
+const NodePolyfillPlugin = require("node-polyfill-webpack-plugin");
+
+/** @type {import("webpack").Configuration} */
+module.exports = {
+    mode: "none",
+    entry: {
+        no_unsanitize: ["./web.js"],
```

Comment:
Question:

Its unfortunate we have to defined a separate entry point file web.js, but based on the eslint API docs it seems that while the ESLint class expects a plugin config property to associate plugin name with its exported object, but that is preprocessed internally before passing it to the Linter class and so Linter defineRules method seems the best workaround available at the moment (at least based on a brief digging into eslint internals it appears to be).

Still, I’m wondering (and so I wanted to ask this explicitly) if as an alternative it may be reasonable to export a new `registerRulesToLinter` method from the index.js file, which given a `Linter` instance would be taking care of registering the rules with the right namespace prefixed to each plugin rule id, e.g. by adding to index.js something like:

```
registerRulesToLinter(linter) { 
    for (const ruleId of Object.keys(this.rules)) {     
        linter.defineRule(`no-unsanitized/${ruleId}`, this.rules[ruleId]);
         }
},
```

And then using it like:

```
const linter = window.linter = new eslint.Linter();
no_unsanitized.registerRulesToLinter(linter);
```

wdyt?

Reply:
I am not sure about it, but I would absolutely hate to keep track of eslint internals and depend on them. I'd rather stick with their public API.

Reply:
100% agree. I'm very OK with a level of indirection if this makes maintenance less painful.